writer.go

```diff
@@ -159,9 +156,9 @@ func parseRecordPayload(payload sdk.Change) (parsed recordPayload, err error) {
 	return parsed, nil
 }
 
-func recordMetadata(data sdk.Metadata) (*pinecone.Metadata, error) {
+func parseRecordMetadata(rec sdk.Record) (*pinecone.Metadata, error) {
```

Comment:
In this case, I believe it's useful to name the method after what's produced, i.e. Pinecone metadata.